### PR TITLE
Support for Conditional Compilation

### PR DESCRIPTION
**Pull request by vannuysm**
Add support for conditional compilation as per #60

### CODE REVIEW

**Comment by nishtahir** [comment]
A working sample needs to go into the `sample-project`

File: grammar/BrightScript.g4 [hunk 1]
```diff
@@ -50,6 +51,15 @@ associativeElementInitializer
     : (identifier | reservedWord | stringLiteral) COLON assignableExpression
     ;
 
+conditionalCompilationStatement
+    : CONDITIONAL_CONST untypedIdentifier EQUALS expression
+    | CONDITIONAL_ERROR .*?
+    | CONDITIONAL_IF expression
+    | CONDITIONAL_ELSEIF expression
+    | CONDITIONAL_ELSE expression
```
Comment (loopyDeveloper) [inline grammar/BrightScript.g4]: @vannuysm 

maybe else should not have an expression?

File: grammar/BrightScript.g4 [hunk 2]
```diff
@@ -558,6 +568,32 @@ WS
...
 
+CONDITIONAL_CONST
+    : '#' C O N S T
+    ;
+
+CONDITIONAL_ELSE
+    : '#' ELSE
+    ;
+
+CONDTIONAL_ELSEIF
```
Comment (loopyDeveloper) [inline grammar/BrightScript.g4]: typo spelling here

should be CONDITIONAL_ELSEIF

File: grammar/BrightScript.g4 (hunk 1 above)
Comment (vannuysm) [reply grammar/BrightScript.g4]: Good catch!

File: grammar/BrightScript.g4 (hunk 2 above)
Comment (vannuysm) [reply grammar/BrightScript.g4]: Some day I'll make my 8th grade typing teacher proud. Today is not that day. 😃

**Comment by vannuysm** [comment]
I have added a couple of test scenarios to the sample project and fixed the typos and such. I feel comfortable enough to remove the `DO NOT MERGE` label from this now.

**Pull request merged**